dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -161,7 +161,24 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
             THROW_HR_IF(E_INVALIDARG, taskClsid == GUID_NULL);
 
             auto registrationOptions = details.Options();
-            THROW_HR_IF(E_INVALIDARG, WI_AreAllFlagsClear(registrationOptions, PushNotificationRegistrationOptions::PushTrigger | PushNotificationRegistrationOptions::ComActivator));
+
+            auto isBackgroundTaskFlagSet{ WI_IsAnyFlagSet(registrationOptions, PushNotificationRegistrationOptions::PushTrigger | PushNotificationRegistrationOptions::ComActivator) };
+            THROW_HR_IF(E_INVALIDARG, isBackgroundTaskFlagSet && !IsActivatorSupported(registrationOptions));
+
+            auto isProtocolActivatorSet{ WI_IsFlagSet(registrationOptions, PushNotificationRegistrationOptions::ProtocolActivator) };
+            THROW_HR_IF(E_INVALIDARG, isProtocolActivatorSet && !IsActivatorSupported(registrationOptions));
+
+            if (isProtocolActivatorSet)
+            {
+                auto coInitialize = wil::CoInitializeEx();
+
+                wil::com_ptr<INotificationsLongRunningPlatform> notificationPlatform{
+                    wil::CoCreateInstance<NotificationsLongRunningPlatform, INotificationsLongRunningPlatform>(CLSCTX_LOCAL_SERVER) };
+
+                wil::unique_cotaskmem_string processName;
+                THROW_IF_FAILED(GetCurrentProcessPath(processName));
+                THROW_IF_FAILED(notificationPlatform->RegisterActivator(processName.get()));
+            }
 
             DWORD cookie = 0;
```

Comment:
Why not put the whole block below in the else clause? It's not relevant to protocol right?